manifests/server.pp

```diff
@@ -83,7 +83,7 @@
 # @param extra_systemd_config Adds extra config to systemd config file, can for instance be used to add extra openfiles. This can be a multi line string
 #
 class postgresql::server (
-  $postgres_password                               = undef,
+  Optional[Variant[String, Sensitive[String]]] $postgres_password = undef,
```

Comment:
Just thinking about compatibility here: could a user specify an integer password?

Reply:
You mean Compatibility to `postgresql::postgresql_password`?
```
    required_param 'Variant[String[1], Sensitive[String[1]], Integer]', :password
```
I don't see `Integer` as valid there neither.

Reply:
Due to other work I didn't have time to look into this for a while. How is it not valid?

Reply:
In my Understanding a Password is a String, not an Integer.  But anyway, I adapted your Suggestion:  

```
Optional[Variant[String[1], Sensitive[String[1]], Integer]] $postgres_password = undef,
```

Reply:
Just for your understanding, you can have a numeric password and if you don't quote it in YAML (with Hiera for example) it becomes `Integer`. I agree it's not that likely, but it can happen and we should be consistent.